### DISCUSSION TITLE
[Help] Linux Client not working?

### DISCUSSION THREAD

**Discussion by tm6907** [cmu-sei/GHOSTS#327] [Q&A] [answered]
I'm looking to add a linux client to my Ghosts setup, got a Win10 working fine thanks to you help but this Ubuntu setup doesn't want to play along. 
<img width="1499" alt="image" src="https://github.com/cmu-sei/GHOSTS/assets/162140616/148c3992-db7b-4b73-b515-c20c3d741174">
Running the client .dll file it starts and connects then just halts and never progresses. Nothing Indicating any errors in the log files either. Also tried on a Rocky 8 box and get the same thing. 
<img width="1499" alt="image" src="https://github.com/cmu-sei/GHOSTS/assets/162140616/29f89904-d35e-49e1-bb39-6a54731327d2">

**Comment by sei-dupdyke** [comment, marked as answer]
I just published a new binary 8.0.7 that should fix this issue, sorry for the confusion.

**Reply by tm6907** [reply]
Ok, tried it out and got some progress but still broken. 
<img width="1022" alt="image" src="https://github.com/cmu-sei/GHOSTS/assets/162140616/b5b1c8cc-6343-4e01-920c-441b8d912027">
Same stat up as before, says its connected then waits abit then this error over and over then aborts. Sorry not sure how to fix it

**Reply by sei-dupdyke** [reply]
Just so I am clear — are you running the API or just running this client standalone?

Also, does the logs/app.log have any errors in it that might help us to troubleshoot?

**Reply by tm6907** [reply]
This is just running the client on Ubuntu, the API is set up on a separate VM. Sometimes it just halts and never progresses, the other times it aborts with all those errors as before. Nothing in the logs weirdly. 
<img width="1502" alt="image" src="https://github.com/cmu-sei/GHOSTS/assets/162140616/6e566f01-7f37-4bc8-ade0-761d7d7d11ec">

**Comment by sei-dupdyke** [comment]
Weird — the client works fine for me, connected to the API or not. Can I see your timeline.json file?

**Reply by tm6907** [reply]
Yeah not sure whats happening, I had a working linux client on v7 but not here. The timeline.json file is unedited 
<img width="1502" alt="image" src="https://github.com/cmu-sei/GHOSTS/assets/162140616/b30f8c87-9a8b-458a-8779-3e424554d777">
<img width="1502" alt="image" src="https://github.com/cmu-sei/GHOSTS/assets/162140616/faf239e0-3414-4fb4-b16a-963c26c0e74e">

The only other changes are to the application.json where I've added the IP to the API machine and enabled Health check.

**Reply by sei-dupdyke** [reply]
Remove the UTCTimeBlocks line please and restart the client.

**Reply by tm6907** [reply]
Tried and same issue :/

**Reply by tm6907** [reply]
Its not essential I get a linux client working, I do have two working win10 clients now but I have some questions about the swaggerUI

**Reply by sei-dupdyke** [reply]
@tm6907 - please see #334 - I wonder if this is what you are seeing?

**Reply by tm6907** [reply]
Don't think so, I've tried the new version you've pushed and no change :/

**Reply by sei-dupdyke** [reply]
Is this an amd64 machine or arm?

**Reply by tm6907** [reply]
x86_64 architecture so amd64

**Comment by tm6907** [comment]
@sei-dupdyke Ok made some progress, this is on the latest linux client with the latest geckodriver for linux64...
Running the geckodriver alone runs fine but when running the client .dll it restarts the driver over and over. Any idea of the issue?

![image](https://github.com/cmu-sei/GHOSTS/assets/162140616/36748afd-2e5e-4a9a-9a74-1f8e70baa748)

![image](https://github.com/cmu-sei/GHOSTS/assets/162140616/9e4d4bdc-5f8b-47b9-9f62-069f28ca2268)

It has at least checked into the API now, so near the end. Thanks in advance

**Reply by sei-dupdyke** [reply]
I believe we can run geckodriver independently, but if there is a version mismatch between it and FF, that's where the problems come in — what is the version of FF and what is the version of GD? Do they match up?

**Reply by tm6907** [reply]
FF is 125.0.3 and geckodriver is 0.34.0 so looks like both are the latest

**Reply by tm6907** [reply]
Got it working on Rocky linux, thats with FF version 115.10.0esr and the same geckodriver

**Comment by sei-dupdyke** [comment]
Created #352 to research this further, sorry for the hassle, but thank you for researching.
